### PR TITLE
Fix top level completions #35400

### LINKED ISSUE #35400: Completions are not provided immediately after the annotation declaration
**Description:**
$title and refer to the following source, where the completions are not provided at the cursor

```

import ballerina/module1;

annotation testAnnotation on class;
<cursor>


function testFunction() {

}

```

**Affected Versions:**
2201.0.0 at least

### PR DESCRIPTION
**Pull request by nadeeshaan**
## Purpose
Fixes #35400

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/AnnotationDeclarationNodeContext.java [hunk 1]
```diff
@@ -91,10 +91,12 @@ public AnnotationDeclarationNodeContext() {
...
     @Override
     public boolean onPreValidation(BallerinaCompletionContext context, AnnotationDeclarationNode node) {
-        Token token = node.annotationKeyword();
+        Token annotationKeyword = node.annotationKeyword();
+        Token semicolonToken = node.semicolonToken();
         int cursor = context.getCursorPositionInTree();
         
-        return !token.isMissing() && cursor >= token.textRange().endOffset();
+        return !annotationKeyword.isMissing() && cursor >= annotationKeyword.textRange().endOffset()
+                && (semicolonToken.isMissing()) || cursor < semicolonToken.textRange().endOffset();
```
Comment (TharushiJay) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/AnnotationDeclarationNodeContext.java:99]:
```suggestion
                && (semicolonToken.isMissing() || cursor < semicolonToken.textRange().endOffset());
```

Comment (nadeeshaan) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/AnnotationDeclarationNodeContext.java]: @TharushiJay you are correct. Updated.

File: language-server/modules/langserver-core/src/test/resources/completion/annotation_decl/source/source21.bal [hunk 2]
```diff
@@ -0,0 +1,10 @@
+
+import ballerina/module1;
+
+annotation testAnnotation on class;
+
+
+
```
Comment (gayalkuruppu) [inline language-server/modules/langserver-core/src/test/resources/completion/annotation_decl/source/source21.bal:7]: Are these lines required? line 1, 6,  and 7?

Comment (nadeeshaan) [reply language-server/modules/langserver-core/src/test/resources/completion/annotation_decl/source/source21.bal:7]: This is added intentionally

**Review by mohanvive — APPROVED** [review]

**Pull request merged**